### DISCUSSION TITLE
Using uv3dp to export file formats: all formats are fine except CWS, SL1 and ZCODEX

### DISCUSSION THREAD

**Discussion by Megidd** [sn4k3/UVtools#223] [Q&A] [answered]
I'm using [this Go package](https://github.com/ezrec/uv3dp) to export a collection of layer images to popular file formats. UVtools can open all the exported formats except:

* CWS
* SL1
* ZCODEX

[bad-formats.zip](https://github.com/sn4k3/UVtools/files/6680596/bad-formats.zip)

The three bad formats are attached.

Considering that I'm not making any modifications to the [Go package](https://github.com/ezrec/uv3dp), I wonder what could be the cause?

Thanks!
😃 

-----------------------

## Error screenshots

![2021-06-19 16_14_34-Window](https://user-images.githubusercontent.com/17475482/122641339-df8ed700-d119-11eb-9208-27374c160626.png)

![2021-06-19 16_15_46-Window](https://user-images.githubusercontent.com/17475482/122641342-e74e7b80-d119-11eb-9c96-9cda4e239f9c.png)

![2021-06-19 16_16_27-](https://user-images.githubusercontent.com/17475482/122641346-ecabc600-d119-11eb-884d-076ae8a187d0.png)

**Comment by sn4k3** [comment, marked as answer]
**CWS:** slice.conf is missing, UVtools read global information from that file, printer does not require that file but it's generated with all slicers and i enforce that verification too.
**SL1:** prusaslicer.ini is missing, on first versions of sl1 that file didn't exist but now it is and it's critical because is where uvtools get all information from.
**Zcodex:** my MaterialId is uint, i must change to string... uv3dp is right

You can open a request @uv3dp to complete the fileformats with that extra missing files to add compability between UVtools.
You can also attempt a fork, and a pull request on the end, will not be hard to complete that.
